types/index.d.ts

```diff
@@ -342,20 +342,69 @@ export interface CookiesOptions {
 }
 
 /** [Documentation](https://next-auth.js.org/configuration/events) */
-export type EventType =
-  | "signIn"
-  | "signOut"
-  | "createUser"
-  | "updateUser"
-  | "linkAccount"
-  | "session"
-  | "error"
+export type EventCallback<MessageType = unknown> = (
+  message: MessageType
+) => Promise<void>
 
-/** [Documentation](https://next-auth.js.org/configuration/events) */
-export type EventCallback = (message: any) => Promise<void>
+/**
+ * If using a `credentials` type auth, the user is the raw response from your
+ * credential provider.
+ * For other providers, you'll get the User object from your adapter, the account,
+ * and an indicator if the user was new to your Adapter.
+ */
+export interface SignInEventMessage<TUser> {
+  user: TUser
+  account: Record<string, unknown>
+  isNewUser?: boolean
+}
 
-/** [Documentation](https://next-auth.js.org/configuration/events) */
-export type EventsOptions = Partial<Record<EventType, EventCallback>>
+export interface LinkAccountEventMessage<TUser> {
+  user: TUser
+  providerAccount: Record<string, unknown>
+}
+
+/**
+ * The various event callbacks you can register for from next-auth
+ */
+export interface EventCallbacks<
+  TSignOutMessage = unknown,
+  TSessionMessage = unknown
+> {
+  signIn: EventCallback<SignInEventMessage<User>>
+  signOut: EventCallback<TSignOutMessage>
+  createUser: EventCallback<User>
+  updateUser: EventCallback<User>
+  linkAccount: EventCallback<LinkAccountEventMessage<User>>
+  session: EventCallback<TSessionMessage>
+  error: EventCallback
+}
+
+/**
+ * The event callbacks will take this form if you are using JWTs:
+ * signOut will receive the JWT and session will receive the session and JWT.
+ */
+export type JWTEventCallbacks = EventCallbacks<
+  JWT & Record<string, any>,
+  {
+    session: Session & Record<string, any>
+    jwt: JWT
+  }
+>
+
+/**
+ * The event callbacks will take this form if you are using Sessions
+ * and not using JWTs:
+ * signOut will receive the underlying db adapter's session object, and session
+ * will receive the NextAuth client session with extra data.
+ */
+export type SessionEventCallbacks = EventCallbacks<
+  Session | null,
+  {
+    session: Session & Record<string, any>
```

Comment:
Here and in line `387` we don't need `Record<string, any>`, these interfaces can be extended through [module augmentation](https://next-auth.js.org/getting-started/typescript) as explained before 👍🏽

Reply:
Done 👍